architecture-decision-record/0033-s3-state-bucket-condition-security.md

```diff
@@ -0,0 +1,47 @@
+# 33.  Increase security of sensitive S3 objects (state bucket)
+
+Date: 2024-09-06
+
+## Status
+
+❌ Rejected
+
+## Context
+
+While we can control access to objects in S3 such as terraform state files through bucket policies, we can risk accidental disclosure if bucket policies & IAM policies are not suitably strict.
+
+Investigated [here](https://github.com/ministryofjustice/modernisation-platform/issues/5346).
+
+It solves the ability for a user to move laterally across roles like so:
+
+- User in Account B assumes Role X in Account A that has access to Object B
+- User then assumes Role Y that has access to Object C
+- Because the aws:PrincipalTag remains consistent as that of Account B, this will not match the value in the object tag and access to Object C will be denied
+
+This has been rectified since the ticket was raised.
+
+## Options
+
+- Adding condition to policy [here](https://github.com/ministryofjustice/modernisation-platform/pull/7860/files)
```

Comment:
Can this be amended to reference a file in a specific commit hash rather than PR please? That's more `git` appropriate than linking to a PR that may be closed.